### ISSUE TITLE
Support completion for inherited methods in traits

### ISSUE THREAD

**Issue by kirillkh** [intellij-rust/intellij-rust#911]
```
trait A {
    fn a(&self);
}

trait B: A {
    fn b(&self) {
        self.a(); // <-- code completion doesn't work after "self."
    }
}

struct X;

// impl A for X {
//    fn a(&self) {}
// }

impl B for X {
    fn b(&self) {
        self.a(); // <-- code completion doesn't work after "self." 
                  // However, it DOES work if you uncomment the "impl A" block.
                  // Apparently, code completion is based on "impl" blocks and not on traits?
    }
}
```

**Comment by matklad** [comment]
First case is done in https://github.com/intellij-rust/intellij-rust/commit/3e82d5ce7b4b2e65e2cc84e2bd2c72abd45f968a

**Comment by matklad** [comment]
Something similar to the second case is done in https://github.com/intellij-rust/intellij-rust/commit/f4386de99c446fae6d3bf391ab0b05c35a9b7d98

I am not entirely sure that the second case should work as is though: the compiler would complain that ``the trait `A` is not implemented for `X` `` so in some sense it is correct that we don't find the implementation for `A` as well.

Thanks for the report!

**Comment by kirillkh** [comment]
Is plugin version 0.1.0.1502 supposed to have contain this fix? Because it's not working for me.

**Comment by matklad** [comment]
>Is plugin version 0.1.0.1502 supposed to contain this fix? Because it's not working for me.

No, this will be in alpha next monday. 

It's available in nightly since 0.1.0.1509-nightly: https://plugins.jetbrains.com/idea/plugin/8182-rust/31832-update

**Comment by kirillkh** [comment]
Thanks for a quick fix, confirmed.
